Source/WTF/wtf/Assertions.cpp

```diff
@@ -77,7 +77,7 @@ ALLOW_NONLITERAL_FORMAT_BEGIN
 
 #if USE(CF)
     if (strstr(format, "%@")) {
-        auto cfFormat = adoptCF(CFStringCreateWithCString(kCFAllocatorDefault, format, kCFStringEncodingUTF8));
+        auto cfFormat = adoptCF(CFStringCreateWithCStringNoCopy(kCFAllocatorDefault, format, kCFStringEncodingUTF8, kCFAllocatorNull));
```

Comment:
Is it documented that `CFStringCreateWithFormatAndArguments` is guaranteed to not retain the format string? Or are we just guessing based on common sense?

Reply:
It's not guaranteed but it doesn't matter because format string isn't going anywhere.

Reply:
I have no idea what distinction you are drawing. Saying “format string isn't going anywhere” is saying you are guessing.

Reply:
Here’s an example of how this could work in a surprising way: Imagine a version that cached the format string to optimize using the same format string multiple times and detecting if it’s the same. We need some guarantee that it doesn’t do something like that.

Reply:
> I have no idea what distinction you are drawing. Saying “format string isn’t going anywhere” is saying you are guessing.

I'm not guessing. Whatever format is pointing to is going to remain there for the entire lifetime of the CFString because it lasts until the end of the function.

That's a fact. It makes no sense for this to change between it being used as the backing store and when the CFString is released.

Reply:
There were in fact cases of this I considered and then immediately changed back because it was found that it was actually better for the string to be copied unconditionally.